dataInterface/items.js

```diff
@@ -9,20 +9,27 @@ module.exports.getAll = () => {
 }
 
 module.exports.getById = (itemId) => {
-  // TODO: complete writing this function
+  return module.exports.items.find(item => item.id == itemId)
 }
 
 module.exports.deleteById = async (itemId) => {
-    // TODO: complete writing this function
+  const index = module.exports.items.findIndex((item) => item.id == itemId)
+  let test = module.exports.items.splice(index, 1)
+  return module.exports.items
+
 }
 
 module.exports.updateById = async (itemId, newObj) => {
-    // TODO: complete writing this function
+  let item = module.exports.items.find(item => item.id == itemId)
+  if(item){
+    item.field = newObj.field
+    return module.exports.items
+  }
 }
 
 module.exports.create = async (item) => {
   const id = uuid.v4();
-  const newItem = { ...item, id };
+  const newItem = { id, ...item };
```

Comment:
`id` should be the second argument, as that will overwrite any `id` field that might have been sent as part of the new object. We want the new object's id field to be the newly-generated unique id from line 31.

![Screen Shot 2022-07-16 at 3 52 01 PM](https://user-images.githubusercontent.com/2523812/179374214-8cc6035c-cded-4a5d-8618-f4dcfa35cb13.png)
.